pkgs/by-name/eb/ebpf-usb/package.nix

```
@@ -0,0 +1,47 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, bcc
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ebpf-usb";
+  version = "unstable-2022-04-03";
+  pyproject = false;
+
+  src = fetchFromGitHub {
+    owner = "francisrstokes";
+    repo = "ebpf-usb";
+    rev = "3ab6f0d8c6ece51bbb5cc5e05daa4008eccd70e8";
+    hash = "sha256-n3ttFej9sroTqAOgyAejwKT+aMt/z7HlVPV6CVGPNUQ=";
+  };
+
+  makeWrapperArgs = [
+    "--set PYTHONUNBUFFERED 1"
+  ];
+
+  pythonPath = with python3Packages; [
+    bcc
+    hexdump
+  ];
```

Since `bcc` is not a Python package, do this:

```suggestion
  pythonPath = [ bcc ] ++ (with python3Packages; [
    hexdump
  ]);
```

---

Quick explanation: with is a dangerous keyword!

Longer explanation: `with` is a dangerous keyword, it brings all the universe of  `python3Packages` into the scope of this list.

If something in the list is inside python3Packages, then it will be selected with higher priority.
It is what you want for `hexdump` I believe.

If something is not inside, then it will be selected elsewhere.

Now suppose that in the future a new Python package called `bcc` is created and packaged.

Then your expression will fail miserably with (pun not intended) a heisenbug, since it will select bcc from the python3Packages instead of outside it.

---

Good to know!